### ISSUE TITLE
Triggered push_redirect by PubSub Broadcast does not work

### ISSUE THREAD

**Issue by Menkir** [phoenixframework/phoenix_live_view#1311]
### Environment

* Elixir version (elixir -v): 1.10.4 (compiled with Erlang/OTP 21)
* Phoenix version (mix deps): 1.5.7
* Phoenix LiveView version (mix deps): 0.15.3
* NodeJS version (node -v): v12.14.0
* NPM version (npm -v): 6.13.4
* Operating system: ubuntu 20
* Browsers you attempted to reproduce this bug on (the more the merrier): chrome, firefox
* Does the problem persist after removing "assets/node_modules" and trying again? Yes

### Actual behavior
Whether this is a bug or incorrect use I cannot say clearly, but: 

I have two LiveViews per path. The first LiveView is statically rendered and always available so it's rendered in root layout. The first LiveView contains a button that changes colour after a click and broadcasts a message via PubSub. The second LiveView is mounted on the router. I would now like to trigger a push_redirect from the first LiveView so that the second LiveView updates itself. What happens instead is that the push_redirect takes place, but the state of the first LiveView is not updated. The strange thing here is that the state shown correctly in a log output.

Here is a working example to reproduce the behaviour: [example](https://github.com/Menkir/Myapp)

**Comment by chrismccord** [comment]
I believe this is working as intended. We do not replace a LiveView on the page in the route layout as it is isolated on its own outside of the live layout. To control remounting of the layout LiveView (but giving it a dynamic ID), you need to move it into the live layout. Thanks!

**Comment by Menkir** [comment]
Thanks @chrismccord for the reply,
I've moved the outer liveview to the live layout instead of the root layout but it still re-mounts. I am not sure I have explained my problem accurately.
I don't want to remount the outer LiveView when I go to another page. I want the outer LiveView to keep its state when I change the page. 
So according to the doc, a push_redirect should only replace the current liveView in @inner_content. Everything outside of it is not reloaded. This means that the outer liveview in the root layout outside @inner_content should not be remounted. But it happens nevertheless. Why ?
